### PR TITLE
Add support for GitHub Actions

### PR DESCRIPTION
**Pull request by marcphilipp**
Since JUnit is using GitHub Actions.

### CODE REVIEW

**Comment by facewindu** [comment]
@marcphilipp can you share a link with the sys prop declared by GH actions ?

**Comment by facewindu** [comment]
ah, nvm, found https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#default-environment-variables

File: guided-trials-default-custom-user-data/default-custom-user-data.gradle [hunk 1]
```diff
@@ -135,6 +135,17 @@ void addCiMetadata() {
...
     }
+
+    if (isGitHubActions()) {
+        def repo = System.getenv('GITHUB_REPOSITORY')
+        def runId = System.getenv('GITHUB_RUN_ID')
+        if (repo && runId) {
+            buildScan.link 'CI job', "https://github.com/$repo/actions/runs/$runId"
+        }
+        if (System.getenv('GITHUB_HEAD_REF')) {
+            buildScan.tag 'Fork'
```
Comment (facewindu) [inline guided-trials-default-custom-user-data/default-custom-user-data.gradle]: should we add a custom value with the 'value' of 'GITHUB_HEAD_REF', which is the 'The branch of the head repository.' ?

File: guided-trials-default-custom-user-data/default-custom-user-data.gradle [hunk 2]
```diff
@@ -135,6 +135,17 @@ void addCiMetadata() {
             buildScan.value 'CI agent ID', agentId
         }
     }
+
+    if (isGitHubActions()) {
+        def repo = System.getenv('GITHUB_REPOSITORY')
+        def runId = System.getenv('GITHUB_RUN_ID')
+        if (repo && runId) {
```
Comment (facewindu) [inline guided-trials-default-custom-user-data/default-custom-user-data.gradle]: is this a precaution or are those two variables always set when using GH actions ?

File: guided-trials-default-custom-user-data/default-custom-user-data.gradle [hunk 3]
```diff
@@ -135,6 +135,17 @@ void addCiMetadata() {
...
+            buildScan.link 'CI job', "https://github.com/$repo/actions/runs/$runId"
+        }
+        if (System.getenv('GITHUB_HEAD_REF')) {
+            buildScan.tag 'Fork'
+        }
+    }
```
Comment (facewindu) [inline guided-trials-default-custom-user-data/default-custom-user-data.gradle]: can we change the things to be more like we do for e.g. Jenkins ?

buildScan.link 'GitHub actions build', "https://github.com/$repo/actions/runs/$runId"

buildScan.value 'CI build number', System.getenv('GITHUB_RUN_NUMBER')

// do sth with GITHUB_ACTION
def ghActionLabel = 'GitHub action'
def ghActionName = System.getenv('GITHUB_ACTION')
buildScan.value ghActionLabel, ghActionName
addCustomValueSearchLink 'GitHub action build scans', [(ghActionLabel): ghActionName]

// Do sth with the GITHUB_WORKFLOW
def ghActionWorkflowLabel = 'GitHub workflow'
def ghActionWorkflowName = System.getenv('GITHUB_WORKFLOW')
buildScan.value ghActionWorkflowLabel, ghActionWorkflowName
addCustomValueSearchLink 'GitHub action build scans', [(ghActionWorkflowLabel): ghActionWorkflowName]



=> not totally familiar with GH actions, but would those be useful ?

File: guided-trials-default-custom-user-data/default-custom-user-data.gradle (hunk 1 above)
Comment (marcphilipp) [reply guided-trials-default-custom-user-data/default-custom-user-data.gradle]: Sure, we can do that. What would we call the key?

File: guided-trials-default-custom-user-data/default-custom-user-data.gradle (hunk 2 above)
Comment (marcphilipp) [reply guided-trials-default-custom-user-data/default-custom-user-data.gradle]: Yes, just a precaution. Do you think it's unnecessary?

File: guided-trials-default-custom-user-data/default-custom-user-data.gradle (hunk 3 above)
Comment (marcphilipp) [reply guided-trials-default-custom-user-data/default-custom-user-data.gradle]: I meant to do this but misunderstood what `CI job` does for Bamboo and CircleCI.

I think it should be "GitHub actions run" for the first link. `GITHUB_ACTION` is just a number, I don't think that's very useful, but `GITHUB_WORKFLOW` might. I'll change it accordingly.

File: guided-trials-default-custom-user-data/default-custom-user-data.gradle (hunk 2 above)
Comment (facewindu) [reply guided-trials-default-custom-user-data/default-custom-user-data.gradle]: let's keep it. This is used in customers projects.

File: guided-trials-default-custom-user-data/default-custom-user-data.gradle [hunk 4]
```diff
@@ -135,6 +135,23 @@ void addCiMetadata() {
...
         }
     }
+
+    if (isGitHubActions()) {
+        def repo = System.getenv('GITHUB_REPOSITORY')
+        def runId = System.getenv('GITHUB_RUN_ID')
+        if (repo && runId) {
+            buildScan.link 'GitHub Actions run', "https://github.com/$repo/actions/runs/$runId"
+        }
+        if (System.getenv('GITHUB_HEAD_REF')) {
```
Comment (facewindu) [inline guided-trials-default-custom-user-data/default-custom-user-data.gradle]: wrong copy paste ?
should be "if GITHUB_WORKFLOW"

File: guided-trials-default-custom-user-data/default-custom-user-data.gradle (hunk 1 above)
Comment (facewindu) [reply guided-trials-default-custom-user-data/default-custom-user-data.gradle]: Forked from ?

File: guided-trials-default-custom-user-data/default-custom-user-data.gradle (hunk 3 above)
Comment (facewindu) [reply guided-trials-default-custom-user-data/default-custom-user-data.gradle]: yeah, hard to check without using GH actions
but if you checked on a real usage, then I'm ok

File: guided-trials-default-custom-user-data/default-custom-user-data.gradle (hunk 4 above)
Comment (marcphilipp) [reply guided-trials-default-custom-user-data/default-custom-user-data.gradle]: 🤦‍♂

File: guided-trials-default-custom-user-data/default-custom-user-data.gradle (hunk 1 above)
Comment (marcphilipp) [reply guided-trials-default-custom-user-data/default-custom-user-data.gradle]: I've decided to remove it for now as I don't see a real use case.

File: guided-trials-default-custom-user-data/default-custom-user-data.gradle (hunk 4 above)
Comment (marcphilipp) [reply guided-trials-default-custom-user-data/default-custom-user-data.gradle]: Fixed.

**Comment by marcphilipp** [comment]
We should squash the commits before we merge, either via the GitHub UI or manually (please let me know).

**Comment by marcphilipp** [comment]
> please squash and merge

I suppose this is meant for @etiennestuder as I don't have push permissions?

**Pull request merged**